### PR TITLE
coqPackages.hierarchy-builder: 1.0.0 -> 1.1.0

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130023` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>coqPackages.mathcomp-analysis</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>coqPackages.hierarchy-builder</li>
  </ul>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130023` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.mathcomp-analysis</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130023` at 9f563839 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.mathcomp-analysis</li>
  </ul>
</details>

File: pkgs/development/coq-modules/hierarchy-builder/default.nix [hunk 1]
```diff
@@ -5,22 +5,25 @@ with lib; mkCoqDerivation {
   owner = "math-comp";
   inherit version;
   defaultVersion = with versions; switch coq.coq-version [
-    { case = isGe "8.12";         out = "1.0.0"; }
+    { case = isGe "8.12";         out = "1.1.0"; }
     { case = range "8.11" "8.12"; out = "0.10.0"; }
   ] null;
-  release."1.0.0".sha256  = "0yykygs0z6fby6vkiaiv3azy1i9yx4rqg8xdlgkwnf2284hffzpp";
+  release."1.1.0".sha256  = "sha256-spno5ty4kU4WWiOfzoqbXF8lWlNSlySWcRReR3zE/4Q=";
```
Comment (CohenCyril) [inline pkgs/development/coq-modules/hierarchy-builder/default.nix:11]: I'd rather keep both releases...

File: pkgs/development/coq-modules/hierarchy-builder/default.nix [hunk 2]
```diff
@@ -5,22 +5,25 @@ with lib; mkCoqDerivation {
...
+  installFlags = [ "DESTDIR=$(out)" "COQMF_COQLIB=lib/coq/${coq.coq-version}" ];
   extraInstallFlags = [ "VFILES=structures.v" ];
 
   meta = {
-    description = "Coq plugin embedding ELPI.";
-    maintainers = [ maintainers.cohencyril ];
-    license = licenses.lgpl21;
+    description = "High level commands to declare a hierarchy based on packed classes";
+    maintainers = with maintainers; [ cohencyril siraben ];
+    license = licenses.mit;
```
Comment (CohenCyril) [inline pkgs/development/coq-modules/hierarchy-builder/default.nix:27]: well spotted!

File: pkgs/development/coq-modules/mathcomp-analysis/default.nix [hunk 3]
```diff
@@ -8,15 +8,15 @@ let mca = mkCoqDerivation {
   pname = "analysis";
   owner = "math-comp";
 
-  release."0.3.6".sha256 = "0g2j7b2hca4byz62ssgg90bkbc8wwp7xkb2d3225bbvihi92b4c5";
+  release."0.3.9".sha256 = "sha256-uUU9diBwUqBrNRLiDc0kz0CGkwTZCUmigPwLbpDOeg4=";
```
Comment (CohenCyril) [inline pkgs/development/coq-modules/mathcomp-analysis/default.nix:11]: same, please keep past hashes

**Comment by CohenCyril** [comment]
Aside from that, ok to merge

**Comment by siraben** [comment]
@CohenCyril how do you want the constraint to be? I think it should be based on the version of hierarchy-builder since mathcomp-analysis 0.3.9 doesn't work with hierarchy-builder 1.0.0

**Comment by CohenCyril** [comment]
> @CohenCyril how do you want the constraint to be? I think it should be based on the version of hierarchy-builder since mathcomp-analysis 0.3.9 doesn't work with hierarchy-builder 1.0.0

Please keep the constraint as they are, just do not remove unused revisions.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130023` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.mathcomp-analysis</li>
  </ul>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130023` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.mathcomp-analysis</li>
  </ul>
</details>

**Review by CohenCyril — APPROVED** [review]
LGTM, thanks! CC @vbgl

File: pkgs/development/coq-modules/hierarchy-builder/default.nix [hunk 4]
```diff
@@ -5,22 +5,26 @@ with lib; mkCoqDerivation {
...
   ] null;
+  release."1.1.0".sha256  = "sha256-spno5ty4kU4WWiOfzoqbXF8lWlNSlySWcRReR3zE/4Q=";
   release."1.0.0".sha256  = "0yykygs0z6fby6vkiaiv3azy1i9yx4rqg8xdlgkwnf2284hffzpp";
   release."0.10.0".sha256 = "1a3vry9nzavrlrdlq3cys3f8kpq3bz447q8c4c7lh2qal61wb32h";
   releaseRev = v: "v${v}";
 
   nativeBuildInputs = [ which ];
 
-  propagatedBuildInputs = [ coq-elpi ];
+  propagatedBuildInputs = [ coq-elpi coq.ocaml ];
```
Comment (vbgl) [inline pkgs/development/coq-modules/hierarchy-builder/default.nix:18]: Why does OCaml need to be propagated?

Comment (CohenCyril) [reply pkgs/development/coq-modules/hierarchy-builder/default.nix:18]: You are right, I do not think it does need to.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130023` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.mathcomp-analysis</li>
  </ul>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130023` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.mathcomp-analysis</li>
  </ul>
</details>

**Comment by siraben** [comment]
@CohenCyril @vbgl addressed concerns

**Pull request merged**